### PR TITLE
added methods for currency exchange

### PR DESCRIPTION
**Pull request by Naffets**

### CODE REVIEW

File: src/main/java/com/example/DataService/CurrencyService.java [hunk 1]
```diff
@@ -134,5 +139,57 @@ public void updateHistory(){
...
+				toCurrency = null;
+
+			} catch (Exception e) {
+				logger.error(e.toString());
+				return new BigDecimal(-1);
+			}
+		}
+
+		if (from == null || toCurrency == null) {
+			logger.error("There is no such currency");
```
Comment (dzlataric) [inline src/main/java/com/example/DataService/CurrencyService.java]: you can log which currency is missing - just to be more clear

File: src/main/java/com/example/DataService/CurrencyService.java [hunk 2]
```diff
@@ -39,14 +42,16 @@
 	@Value("${coin-api-url}")
 	private String uri;
 	
+	@Value("${coin-api-currency-exchange-url}")
```
Comment (dzlataric) [inline src/main/java/com/example/DataService/CurrencyService.java]: You could have used the same base url and just add different paths where needed - like this you have a lot of config properties which are almost the same

File: src/main/java/com/example/DataService/CurrencyService.java [hunk 3]
```diff
@@ -134,5 +139,57 @@ public void updateHistory(){
...
+		return from.getPriceusd().divide(to.getPriceusd(), 16, RoundingMode.CEILING);		
+	}
+	
+	
+	public BigDecimal currencyExchangeIntoRealCurrency(Currency from, String to) {
+		
+		RealCurrencyDTO toCurrency = new RealCurrencyDTO();
+
+		RestTemplate restTemplate = new RestTemplate();
+		ResponseEntity<String> response = restTemplate.getForEntity(exchangeUri, String.class);
```
Comment (dzlataric) [inline src/main/java/com/example/DataService/CurrencyService.java]: What is this used for? I just see the status code check

Comment (Naffets) [reply src/main/java/com/example/DataService/CurrencyService.java]: It's just to check status code in order to see if the url is valid.

Comment (dzlataric) [reply src/main/java/com/example/DataService/CurrencyService.java]: You should have error handling for every API call. No need to call an endpoint to see if it is valid - that does not tell us anything. We can discuss this together and i will try to explain as much as i can.

**Review by dzlataric — APPROVED** [review]

**Pull request merged**